File: Infrastructure.Tests/Factories/CashgameFactoryTests.php
namespace tests\FactoryTests{

	use DateTime;
	use entities\CashgameFactoryImpl;
	use entities\CashgameResult;
	use entities\GameStatus;
	use entities\Player;
	use tests\UnitTestCase;

	class CashgameFactoryTests extends UnitTestCase {

		private $earliestBuyinTime;
		private $earliestCashoutTime;
		private $latestBuyinTime;
		private $latestCashoutTime;

		private $results;
		private $location;
		private $status;
		private $id;

		function __construct(){
			$this->earliestBuyinTime = new DateTime("2010-01-01 01:00:00");
			$this->earliestCashoutTime = new DateTime("2010-01-01 03:00:00");
			$this->latestBuyinTime = new DateTime("2010-01-01 02:00:00");
			$this->latestCashoutTime = new DateTime("2010-01-01 04:00:00");

			$this->results = array();
			$result1 = new CashgameResult();
			$result1->setPlayer(new Player());
			$result1->setBuyinTime($this->earliestBuyinTime);
			$result1->setCashoutTime($this->earliestCashoutTime);
			$this->results[] = $result1;
			$result2 = new CashgameResult();
			$result2->setPlayer(new Player());
			$result2->setBuyinTime($this->latestBuyinTime);
			$result2->setCashoutTime($this->latestCashoutTime);
			$this->results[] = $result2;

			$this->location = 'a';
			$this->status = GameStatus::running;
			$this->id = 1;
		}

		function test_Get_CashgamePropertiesAreSet(){
			$sut = $this->getSut();
			$result = $sut->create($this->location, $this->status, $this->id, $this->results);

			$this->assertIdentical($this->location, $result->getLocation());
			$this->assertIdentical($this->status, $result->getStatus());
			$this->assertIdentical($this->id, $result->getId());
			$this->assertIdentical($this->earliestBuyinTime, $result->getStartTime());
			$this->assertIdentical($this->latestCashoutTime, $result->getEndTime());
			$this->assertIdentical(180, $result->getDuration());
		}

		private function getSut(){
			return new CashgameFactoryImpl();
		}

	}

}

File: Infrastructure.Tests/Factories/CashgameFactoryTests.cs
using System;
using System.Collections.Generic;
using Core.Classes;
using Infrastructure.Factories;
using NUnit.Framework;
using Tests.Common;
using Tests.Common.FakeClasses;

namespace Infrastructure.Tests.Factories{

	public class CashgameFactoryTests : MockContainer
    {
        private DateTime _earliestBuyinTime;
		private DateTime _earliestCashoutTime;
		private DateTime _latestBuyinTime;
		private DateTime _latestCashoutTime;

		private List<CashgameResult> _results;
		private string _location;
		private GameStatus _status;
		private int _id;

        [SetUp]
		public void SetUp(){
			_earliestBuyinTime = DateTime.Parse("2010-01-01 01:00:00");
			_earliestCashoutTime = DateTime.Parse("2010-01-01 03:00:00");
			_latestBuyinTime = DateTime.Parse("2010-01-01 02:00:00");
			_latestCashoutTime = DateTime.Parse("2010-01-01 04:00:00");

            _results = new List<CashgameResult>
                {
                    new FakeCashgameResult
                        (
                            buyinTime: _earliestBuyinTime,
                            cashoutTime: _earliestCashoutTime
                        ),
                    new FakeCashgameResult
                        (
                            buyinTime: _latestBuyinTime,
                            cashoutTime: _latestCashoutTime
                        )
                };

			_location = "a";
			_status = GameStatus.Running;
			_id = 1;
		}

        [Test]
		public void Get_CashgamePropertiesAreSet(){
			var sut = GetSut();
			var result = sut.Create(_location, (int)_status, _id, _results);

			Assert.AreEqual(_location, result.Location);
            Assert.AreEqual(_status, result.Status);
            Assert.AreEqual(_id, result.Id);
            Assert.AreEqual(_earliestBuyinTime, result.StartTime);
            Assert.AreEqual(_latestCashoutTime, result.EndTime);
            Assert.AreEqual(180, result.Duration);
		}

		private CashgameFactory GetSut(){
			return new CashgameFactory(
                GetMock<ICashgameResultFactory>().Object,
                GetMock<ICheckpointFactory>().Object);
		}

	}

}